### PR TITLE
UIU-2714: Implement modal windows for BILL ACTUAL COST and DO NOT BILL

### PR DESCRIPTION
**Pull request by Dmitriy-Litvinenko**
## Purpose
Implement modal windows for BILL ACTUAL COST and DO NOT BILL

## Approach
This components created for actual cost. Behavior of current component will be tested in scope of actual cost feature ( https://issues.folio.org/browse/UIU-1863 and https://issues.folio.org/browse/UIU-2705 ).
For increase code coverage was created separate task https://issues.folio.org/browse/UIU-2727

## Refs
https://issues.folio.org/browse/UIU-2714

### CODE REVIEW

File: src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCostConfirm/ActualCostConfirm.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      label={getTitle()}
+      dismissible
+      open={isOpen}
+      onClose={onKeepEditing}
+      footer={renderFooter}
+    >
+      <Row>
+        <Col xs={12}>
+
+            <Row>
```
Comment (mkuklis) [inline src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCostConfirm/ActualCostConfirm.js:105]: @Dmitriy-Litvinenko would you mind fixing this lint issue?

**Review by mkuklis — APPROVED** [review]
Nice and clean @Dmitriy-Litvinenko

File: src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCostConfirm/ActualCostConfirm.js (hunk 1 above)
Comment (Dmitriy-Litvinenko) [reply src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCostConfirm/ActualCostConfirm.js:104]: Fixed

File: src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCost/ActualCost.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  ACTUAL_COST_TYPES,
+  ACTUAL_COST_RECORD_FIELD_NAME,
+  ACTUAL_COST_RECORD_FIELD_PATH,
+  DEFAULT_VALUE,
+  ACTUAL_COST_MODAL_DEFAULT,
+  ACTUAL_COST_DEFAULT,
+  ACTUAL_COST_PROP_TYPES,
+} from '../../../../../constants';
+
+const ActualCost = ({
```
Comment (artem-blazhko) [inline src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCost/ActualCost.js:33]: I would rename it to `ActualCostModal`. What do you think about it?
In my opinion, the name should say that it is a modal window.

File: src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCostConfirm/ActualCostConfirm.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  ACTUAL_COST_RECORD_FIELD_NAME,
+  ACTUAL_COST_RECORD_FIELD_PATH,
+  ACTUAL_COST_TYPES,
+  ACTUAL_COST_PROP_TYPES,
+  DEFAULT_VALUE,
+} from '../../../../../constants';
+
+import css from './ActualCostConfirm.css';
+
+const ActualCostConfirm = ({
```
Comment (artem-blazhko) [inline src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCostConfirm/ActualCostConfirm.js:29]: I would rename it to `ActualCostConfirmModal`. What do you think about it?
In my opinion, the name should say that it is a modal window.

File: src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/RenderActions/components/PatronDetailsLink.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import {
+  ACTUAL_COST_RECORD_FIELD_NAME,
+  ACTUAL_COST_RECORD_FIELD_PATH,
+  DEFAULT_VALUE,
+} from '../../../../../../constants';
+
+const PatronDetailsLink = ({ actualCostRecord }) => {
+  const userId = get(actualCostRecord, ACTUAL_COST_RECORD_FIELD_PATH[ACTUAL_COST_RECORD_FIELD_NAME.USER_ID], DEFAULT_VALUE);
+  const isPatronDetailsLinkActive = userId;
```
Comment (artem-blazhko) [inline src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/RenderActions/components/PatronDetailsLink.js:16]: Does it make sense to have an additional variable?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Change history for ui-users
 
 ## 8.3.0 IN PROGRESS
+* Implement modal windows for BILL ACTUAL COST and DO NOT BILL. Refs UIU-2714.
 
 * create Jest/RTL test for UserDetail.js. Refs UIU-2421
```
[the remark this answers is not in the document]
Comment (Dmitriy-Litvinenko) [reply CHANGELOG.md:6]: Fixed

File: src/views/LostItems/LostItemsListContainer/components/LostItemsList/LostItemsList.js [hunk 6]
```diff
@@ -130,34 +96,92 @@ const LostItemsList = ({
...
+
+  const lostItemsListFormatter = {
+    [COLUMNS_NAME.PATRON]: (actualCostRecord) => {
+      const patronGroup = get(actualCostRecord, ACTUAL_COST_RECORD_FIELD_PATH[ACTUAL_COST_RECORD_FIELD_NAME.USER_PATRON_GROUP], DEFAULT_VALUE);
+      const patronName = getPatronName(actualCostRecord);
+
+      return (
+        <div>
+          <div>{patronName}</div>
+          <div>{`(${patronGroup})`}</div>
```
[the remark this answers is not in the document]
Comment (Dmitriy-Litvinenko) [reply src/views/LostItems/LostItemsListContainer/components/LostItemsList/LostItemsList.js:111]: Fixed

File: src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCost/ActualCost.js (hunk 2 above)
Comment (Dmitriy-Litvinenko) [reply src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCost/ActualCost.js:33]: Fixed

File: src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCostConfirm/ActualCostConfirm.js (hunk 3 above)
Comment (Dmitriy-Litvinenko) [reply src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/ActualCostConfirm/ActualCostConfirm.js:29]: Fixed

File: src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/RenderActions/components/PatronDetailsLink.js (hunk 4 above)
Comment (Dmitriy-Litvinenko) [reply src/views/LostItems/LostItemsListContainer/components/LostItemsList/components/RenderActions/components/PatronDetailsLink.js:16]: For code readability

File: src/views/LostItems/constants.js [hunk 7]
```diff
@@ -1,7 +1,10 @@
+import PropTypes from 'prop-types';
+
 import {
   itemStatuses,
 } from '../../constants';
 
+export const DEFAULT_VALUE = '';
```
[the remark this answers is not in the document]
Comment (Dmitriy-Litvinenko) [reply src/views/LostItems/constants.js:7]: Yes name is general for use it for all cases as default value

**Comment by Dmitriy-Litvinenko** [comment]
@artem-blazhko Could you please re-review?

**Review by artem-blazhko — APPROVED** [review]

**Pull request merged**